### PR TITLE
Add Dashboard button and icon button component - Closes #6225 button component

### LINKED ISSUE #6225: Create react component for buttons
### Description

Create Button component to show an button with properties
- size
- onClick: () => void

Create IconButton component to show an button with properties
- icon
- onClick: () => void

### Acceptance Criteria

- Use Icon component for IconButton
- Use SCSS constant
- Add sample to the main page for testing

### Additional Information
Related Issues: #6216

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### What was the problem?

This PR resolves #6225 

### How was it solved?

- Added Button component
- Added IconButton component
- Added styling

### How was it tested?

Check mainpage of dashboard

### CODE REVIEW

File: framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/Button.module.scss [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	justify-content: center;
+	align-items: center;
+	padding: 14px 30px;
+	color: $color-white;
+	width: 167px;
+	height: 46px;
+	left: 20px;
+	top: 20px;
+	/* Ultramarine blue */
+	background: #4070F4;
```
Comment (shuse2) [inline framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/Button.module.scss:15]: this should use the constant defined

File: framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/IconButton/index.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+interface Props {
+	name: string;
+	size?: 's' | 'm' | 'l' | 'xl';
+}
+
+const onSubmit = (): void => undefined;
+
+const IconButton: React.FC<Props> = props => {
+	const { name } = props;
+	const size = props.size ?? 'm';
```
Comment (shuse2) [inline framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/IconButton/index.tsx:26]: Icon button should use the icon component

File: framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/IconButton/IconButton.module.scss [hunk 3]
```diff
@@ -0,0 +1,47 @@
+@import '../../styles/constants';
```
Comment (shuse2) [inline framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/IconButton/IconButton.module.scss:1]: this file shouldn't be required if Icon component is used

File: framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/Button.module.scss [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	border-radius: 3px;
```
Comment (shuse2) [inline framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/Button.module.scss:16]: there are several properties that needs to be handled
https://github.com/LiskHQ/lisk-sdk/blob/development/framework-plugins/lisk-framework-faucet-plugin/src/ui/app.module.scss#L146-L169

File: framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/Button.module.scss [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+.button {
+	display: flex;
+	flex-direction: column;
+	justify-content: center;
+	align-items: center;
+	padding: 14px 30px;
+	color: $color-white;
+	width: 167px;
+	height: 46px;
+	left: 20px;
```
Comment (shuse2) [inline framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/Button.module.scss:12]: left/top should be required?

File: framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/index.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	name: string;
+	size?: 's' | 'm' | 'l' | 'xl';
+}
+
+const onSubmit = (): void => undefined;
+
+const Button: React.FC<Props> = props => {
+	const { name } = props;
+	const size = props.size ?? 'm';
+	return <button className={`${styles.button} ${styles[`button-${size}`]}`} onClick={onSubmit}>{name}</button>;
```
Comment (shuse2) [inline framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/index.tsx:27]: name should use the `Text` component

File: framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/Button.module.scss [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+	top: 20px;
+	/* Ultramarine blue */
+	background: #4070F4;
+	border-radius: 3px;
+
+	&.button {
+		&.button-s {
```
Comment (shuse2) [inline framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/Button.module.scss:19]: these are not required i think

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/index.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+const onSubmit = (): void => undefined;
+
+const Button: React.FC<Props> = props => {
+	const { name } = props;
+	const size = props.size ?? 'm';
+	return (
+		<button className={`${styles.button} ${styles[`button-${size}`]}`} onClick={onSubmit}>
+			<Text>{name}</Text>
```
Comment (nazarhussain) [inline framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/index.tsx:30]: Why we need `Text` component here? HTML `button` can wrap the text itself.

File: framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/index.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+	size?: 's' | 'm' | 'l' | 'xl';
+}
+
+const onSubmit = (): void => undefined;
+
+const Button: React.FC<Props> = props => {
+	const { name } = props;
+	const size = props.size ?? 'm';
+	return (
+		<button className={`${styles.button} ${styles[`button-${size}`]}`} onClick={onSubmit}>
```
Comment (nazarhussain) [inline framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/index.tsx:29]: I am not sure why you need to set `onClick` to an empty handler. But if you want to disable any default interaction you should use 

https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault

File: framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/index.tsx [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import * as React from 'react';
+import styles from './Button.module.scss';
+import Text from '../Text';
+
+interface Props {
+	name: string;
```
Comment (nazarhussain) [inline framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/Button/index.tsx:19]: The issue didn't mentioned to have a `name` prop to use as button text. Rather as normal React component `Button` should just render the `children` prop.

File: framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/IconButton/index.tsx [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+	return (
+		<div className={styles['icon-button']} onClick={onSubmit}>
```
Comment (nazarhussain) [inline framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/IconButton/index.tsx:29]: Here you should use the `Button` component and add `Icon` as its children along with `Text`.

File: framework-plugins/lisk-framework-dashboard-plugin/src/ui/pages/MainPage.tsx [hunk 12]
```diff
@@ -34,6 +36,8 @@ const MainPage: React.FC = () => (
 			<Text color="white" type="p">
 				bd81020ded87d21bbfedc45ed...5d90
 			</Text>
+			<Button name={'Button'} size={'m'} />
```
Comment (nazarhussain) [inline framework-plugins/lisk-framework-dashboard-plugin/src/ui/pages/MainPage.tsx:39]: Its usage should be`<Button size={'m'}>Button</Button>

File: framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/IconButton/index.tsx [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import * as React from 'react';
+import Icon from '../Icon';
+import styles from './IconButton.module.scss';
+
+interface Props {
+	name: string;
```
Comment (nazarhussain) [inline framework-plugins/lisk-framework-dashboard-plugin/src/ui/components/IconButton/index.tsx:19]: As per the issue the prop name is `icon` not the `name`.

**Review by nazarhussain — CHANGES_REQUESTED** [review]

**Review by nazarhussain — APPROVED** [review]

**Review by shuse2 — APPROVED** [review]

**Pull request merged**